description: fix gcc argument promotion warning
author: Michael Gilbert <mgilbert@debian.org>

--- a/include/evntrace.h
+++ b/include/evntrace.h
@@ -308,7 +308,7 @@ ULONG WINAPI StartTraceA(PTRACEHANDLE,LP
 ULONG WINAPI StartTraceW(PTRACEHANDLE,LPCWSTR,PEVENT_TRACE_PROPERTIES);
 #define      StartTrace WINELIB_NAME_AW(StartTrace)
 ULONG WINAPI TraceEvent(TRACEHANDLE,PEVENT_TRACE_HEADER);
-ULONG WINAPIV TraceMessage(TRACEHANDLE,ULONG,LPGUID,USHORT,...);
+ULONG WINAPIV TraceMessage(TRACEHANDLE,ULONG,LPGUID,UINT,...);
 ULONG WINAPI TraceMessageVa(TRACEHANDLE,ULONG,LPGUID,USHORT,__ms_va_list);
 ULONG WINAPI UnregisterTraceGuids(TRACEHANDLE);
 
--- a/dlls/ntdll/misc.c
+++ b/dlls/ntdll/misc.c
@@ -585,7 +585,7 @@ ULONG WINAPI EtwTraceMessageVa( TRACEHAN
 /******************************************************************************
  *                  EtwTraceMessage (NTDLL.@)
  */
-ULONG WINAPIV EtwTraceMessage( TRACEHANDLE handle, ULONG flags, LPGUID guid, USHORT number, ... )
+ULONG WINAPIV EtwTraceMessage( TRACEHANDLE handle, ULONG flags, LPGUID guid, UINT number, ... )
 {
     __ms_va_list valist;
     ULONG ret;
