description: ignore array bounds warnings involving ANYSIZE_ARRAY
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/ntdll/Makefile.in
+++ b/dlls/ntdll/Makefile.in
@@ -2,6 +2,7 @@ EXTRADEFS = -D_NTSYSTEM_
 MODULE    = ntdll.dll
 IMPORTLIB = ntdll
 IMPORTS   = winecrt0
+CFLAGS    = -Wno-array-bounds
 EXTRAINCL = $(UNWIND_CFLAGS)
 EXTRALIBS = $(IOKIT_LIBS) $(RT_LIBS) $(PTHREAD_LIBS) $(UNWIND_LIBS)
 EXTRADLLFLAGS = -nodefaultlibs -Wl,--image-base,0x7bc00000
