description: ignore warnings from bison
author: Michael Gilbert <mgilbert@debian.org>

--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -2699,7 +2699,7 @@ static void output_source_y( struct make
     if (find_include_file( make, header ))
     {
         output( "%s: %s\n", obj_dir_path( make, header ), source->filename );
-        output( "\t%s%s -o %s.tab.c -d %s\n",
+        output( "\t%s%s -Wno-conflicts-rr -Wno-conflicts-sr -o %s.tab.c -d %s\n",
                 cmd_prefix( "BISON" ), bison, obj_dir_path( make, obj ), source->filename );
         output( "%s.tab.c: %s %s\n", obj_dir_path( make, obj ),
                 source->filename, obj_dir_path( make, header ));
@@ -2707,7 +2707,7 @@ static void output_source_y( struct make
     }
     else output( "%s.tab.c: %s\n", obj_dir_path( make, obj ), source->filename );
 
-    output( "\t%s%s -o $@ %s\n", cmd_prefix( "BISON" ), bison, source->filename );
+    output( "\t%s%s -Wno-conflicts-rr -Wno-conflicts-sr -o $@ -d %s\n", cmd_prefix( "BISON" ), bison, source->filename );
 }
 
 
