description: add missing format string in calls to fprintf
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/msvcp60/tests/ios.c
+++ b/dlls/msvcp60/tests/ios.c
@@ -1467,7 +1467,7 @@ static void test_istream_tellg(void)
 
         /* filebuf */
         file = fopen(testfile, "wt");
-        fprintf(file, tests[i].str);
+        fprintf(file, "%s", tests[i].str);
         fclose(file);
 
         /* fstream<char> version */
--- a/dlls/msvcp90/tests/ios.c
+++ b/dlls/msvcp90/tests/ios.c
@@ -1969,7 +1969,7 @@ static void test_istream_tellg(void)
 
         /* filebuf */
         file = fopen(testfile, "wt");
-        fprintf(file, tests[i].str);
+        fprintf(file, "%s", tests[i].str);
         fclose(file);
 
         /* fstream<char> version */
