description: fix gcc incompatible pointer warning
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/wininet/internet.c
+++ b/dlls/wininet/internet.c
@@ -1587,7 +1587,7 @@ BOOL WINAPI InternetCrackUrlA(const char
     return ret;
 }
 
-static const WCHAR *url_schemes[] =
+static const wchar_t *url_schemes[] =
 {
     L"ftp",
     L"gopher",
@@ -1622,7 +1622,7 @@ static INTERNET_SCHEME GetInternetScheme
         return INTERNET_SCHEME_UNKNOWN;
 
     for (i = 0; i < ARRAY_SIZE(url_schemes); i++)
-        if (!strncmpiW(lpszScheme, url_schemes[i], nMaxCmp))
+        if (!strncmpiW(lpszScheme, (LPCWSTR)url_schemes[i], nMaxCmp))
             return INTERNET_SCHEME_FIRST + i;
 
     return INTERNET_SCHEME_UNKNOWN;
@@ -4221,7 +4221,7 @@ static LPCWSTR INTERNET_GetSchemeString(
     index = scheme - INTERNET_SCHEME_FIRST;
     if (index >= ARRAY_SIZE(url_schemes))
         return NULL;
-    return url_schemes[index];
+    return (LPCWSTR)url_schemes[index];
 }
 
 /* we can calculate using ansi strings because we're just
