description: fix mismatched types (-Wincompatible-pointer-types)
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/winegstreamer/media_source.c
+++ b/dlls/winegstreamer/media_source.c
@@ -614,7 +614,7 @@ static DWORD CALLBACK read_thread(void *
     struct media_source *source = arg;
     IMFByteStream *byte_stream = source->byte_stream;
     size_t buffer_size = 4096;
-    uint64_t file_size;
+    QWORD file_size;
     void *data;
 
     if (!(data = malloc(buffer_size)))
@@ -1395,7 +1395,7 @@ static HRESULT media_source_constructor(
     UINT64 total_pres_time = 0;
     struct wg_parser *parser;
     DWORD bytestream_caps;
-    uint64_t file_size;
+    QWORD file_size;
     unsigned int i;
     HRESULT hr;
 
