description: fix misleading-indentation warnings
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/oleaut32/vartype.c
+++ b/dlls/oleaut32/vartype.c
@@ -149,15 +149,27 @@ static HRESULT VARIANT_FromDisp(IDispatc
 
 /* Compiler cast where input cannot be negative */
 #define NEGTST(dest, src, func) RETTYP _##func(src in, dest* out) { \
-  if (in < 0) return DISP_E_OVERFLOW; *out = in; return S_OK; }
+  if (in < 0) \
+      return DISP_E_OVERFLOW; \
+  *out = in; \
+  return S_OK; \
+  }
 
 /* Compiler cast where input cannot be > some number */
 #define POSTST(dest, src, func, tst) RETTYP _##func(src in, dest* out) { \
-  if (in > (dest)tst) return DISP_E_OVERFLOW; *out = in; return S_OK; }
+  if (in > (dest)tst) \
+      return DISP_E_OVERFLOW; \
+  *out = in; \
+  return S_OK; \
+  }
 
 /* Compiler cast where input cannot be < some number or >= some other number */
 #define BOTHTST(dest, src, func, lo, hi) RETTYP _##func(src in, dest* out) { \
-  if (in < (dest)lo || in > hi) return DISP_E_OVERFLOW; *out = in; return S_OK; }
+  if (in < (dest)lo || in > hi) \
+      return DISP_E_OVERFLOW; \
+  *out = in; \
+  return S_OK; \
+  }
 
 /* I1 */
 POSTST(signed char, BYTE, VarI1FromUI1, I1_MAX)
--- a/dlls/wininet/internet.c
+++ b/dlls/wininet/internet.c
@@ -2887,9 +2887,10 @@ BOOL WINAPI InternetSetOptionW(HINTERNET
         } else if(dwBufferLength != sizeof(ULONG)) {
             SetLastError(ERROR_INTERNET_BAD_OPTION_LENGTH);
             ret = FALSE;
-        } else
+        } else {
             TRACE("INTERNET_OPTION_ERROR_MASK: %x\n", *(ULONG*)lpBuffer);
             lpwhh->ErrorMask = *(ULONG*)lpBuffer;
+        }
       }
       break;
     case INTERNET_OPTION_PROXY:
