description: fix uninitialized variable compiler warnings
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/comdlg32/filedlg.c
+++ b/dlls/comdlg32/filedlg.c
@@ -247,7 +247,7 @@ static BOOL get_config_key_dword(HKEY hk
 
 static BOOL get_config_key_string(HKEY hkey, const WCHAR *name, WCHAR **value)
 {
-    DWORD type, size;
+    DWORD type, size = 0;
     WCHAR *str;
 
     if (hkey && !RegQueryValueExW(hkey, name, 0, &type, NULL, &size))
--- a/dlls/d3dx9_36/texture.c
+++ b/dlls/d3dx9_36/texture.c
@@ -1301,6 +1301,10 @@ static inline void fill_texture(const st
             case 3: /* Blue */
                 comp_value = value->z;
                 break;
+            default: /* Unhandled */
+                comp_value = 0.f;
+                ERR("Unhandled component, %#x\n", c);
+                break;
         }
 
         if (format->type == FORMAT_ARGBF16)
--- a/dlls/kernel32/locale.c
+++ b/dlls/kernel32/locale.c
@@ -923,7 +923,7 @@ INT WINAPI IdnToUnicode(DWORD dwFlags, L
                         LPWSTR lpUnicodeCharStr, INT cchUnicodeChar)
 {
     INT i, label_start, label_end, out_label, out = 0;
-    WCHAR ch;
+    WCHAR ch = '\0';
 
     TRACE("%x %p %d %p %d\n", dwFlags, lpASCIICharStr, cchASCIIChar,
         lpUnicodeCharStr, cchUnicodeChar);
--- a/dlls/msvcrt/misc.c
+++ b/dlls/msvcrt/misc.c
@@ -272,7 +272,7 @@ static void small_sort(void *base, MSVCR
         int (CDECL *compar)(void *, const void *, const void *), void *context)
 {
     MSVCRT_size_t e, i;
-    char *max, *p;
+    char *max, *p = NULL;
 
     for(e=nmemb; e>1; e--) {
         max = base;
--- a/dlls/msxml3/xdr.c
+++ b/dlls/msxml3/xdr.c
@@ -603,6 +603,10 @@ static xmlNodePtr XDR_E_ElementType(xmlN
                         case ORDER_ONE:
                             xsd_child = xmlNewChild(xsd_base, NULL, xs_all, NULL);
                             break;
+                        default:
+                            xsd_child = NULL;
+                            ERR("Unhandled order enum %d\n", (int)order);
+                            break;
                     }
 
                     FOREACH_CHILD(xdr, xdr_child)
--- a/dlls/ole32/datacache.c
+++ b/dlls/ole32/datacache.c
@@ -2519,7 +2519,7 @@ static HRESULT WINAPI DataCache_EnumCach
     DataCacheEntry *cache_entry;
     int i = 0, count = 0;
     STATDATA *data;
-    HRESULT hr;
+    HRESULT hr = E_FAIL;
 
     TRACE( "(%p, %p)\n", This, enum_stat );
 
--- a/dlls/webservices/reader.c
+++ b/dlls/webservices/reader.c
@@ -2040,7 +2040,7 @@ static inline BOOL is_element_type( unsi
 
 static HRESULT read_attributes_bin( struct reader *reader, WS_XML_ELEMENT_NODE *elem )
 {
-    WS_XML_ATTRIBUTE *attr;
+    WS_XML_ATTRIBUTE *attr = NULL;
     unsigned char type;
     HRESULT hr;
 
@@ -6357,7 +6357,7 @@ static HRESULT read_type_struct( struct
                                  WS_HEAP *heap, void *ret, ULONG size, BOOL *found )
 {
     ULONG i, offset;
-    HRESULT hr;
+    HRESULT hr = E_FAIL;
     char *buf;
 
     if (!desc) return E_INVALIDARG;
--- a/dlls/wined3d/utils.c
+++ b/dlls/wined3d/utils.c
@@ -2659,6 +2659,10 @@ static void check_fbo_compat(struct wine
                     case WINED3D_GL_RES_TYPE_COUNT:
                         color = 0;
                         break;
+                    default:
+                        color = 0;
+                        ERR("Unhandled type enum %d\n", (int)type);
+                        break;
                 }
                 checkGLcall("Post-pixelshader blending check");
 
--- a/dlls/devenum/createdevenum.c
+++ b/dlls/devenum/createdevenum.c
@@ -387,7 +387,7 @@ static void write_filter_data(IPropertyB
     HRESULT hr;
 
     hr = CoCreateInstance(&CLSID_FilterMapper2, NULL, CLSCTX_INPROC, &IID_IAMFilterData, (void **)&fildata);
-    if (FAILED(hr)) goto cleanup;
+    if (FAILED(hr)) goto cleanup2;
 
     hr = IAMFilterData_CreateFilterData(fildata, rgf, &data, &size);
     if (FAILED(hr)) goto cleanup;
@@ -408,8 +408,9 @@ static void write_filter_data(IPropertyB
     if (FAILED(hr)) goto cleanup;
 
 cleanup:
-    VariantClear(&var);
     CoTaskMemFree(data);
+cleanup2:
+    VariantClear(&var);
     IAMFilterData_Release(fildata);
 }
 
--- a/dlls/devenum/mediacatenum.c
+++ b/dlls/devenum/mediacatenum.c
@@ -128,7 +128,7 @@ static HRESULT WINAPI DEVENUM_IPropertyB
     HRESULT res = S_OK;
     LONG reswin32 = ERROR_SUCCESS;
     WCHAR name[80];
-    HKEY hkey;
+    HKEY hkey = NULL;
 
     TRACE("(%p)->(%s, %p, %p)\n", This, debugstr_w(pszPropName), pVar, pErrorLog);
 
--- a/dlls/user32/sysparams.c
+++ b/dlls/user32/sysparams.c
@@ -1835,7 +1835,7 @@ BOOL WINAPI SystemParametersInfoW( UINT
     case SPI_GETNONCLIENTMETRICS:
     {
         LPNONCLIENTMETRICSW lpnm = pvParam;
-        int padded_border;
+        int padded_border = 0;
 
         if (!pvParam) return FALSE;
 
--- a/dlls/mf/session.c
+++ b/dlls/mf/session.c
@@ -425,7 +425,7 @@ static void session_clear_topologies(str
 static void session_set_topo_status(struct media_session *session, struct queued_topology *topology, HRESULT status,
         MF_TOPOSTATUS topo_status)
 {
-    IMFMediaEvent *event;
+    IMFMediaEvent *event = NULL;
     PROPVARIANT param;
 
     if (topo_status == MF_TOPOSTATUS_INVALID)
@@ -2012,7 +2012,9 @@ static HRESULT clock_change_state(struct
             notification = CLOCK_NOTIFY_SET_RATE;
             break;
         default:
-            ;
+            notification = CLOCK_NOTIFY_START;
+            ERR("Unhandled clock_command enum %d\n", (int)command);
+            break;
     }
 
     if (FAILED(hr = clock_call_state_change(system_time, param, notification, clock->time_source_sink)))
--- a/tools/winedump/msc.c
+++ b/tools/winedump/msc.c
@@ -282,7 +282,7 @@ static void do_field(const unsigned char
     const unsigned char*        ptr = start;
     const char*                 cstr;
     const struct p_string*      pstr;
-    int leaf_len, value;
+    int leaf_len, value = 0;
 
     while (ptr < end)
     {
@@ -574,7 +574,7 @@ static void do_field(const unsigned char
 static void codeview_dump_one_type(unsigned curr_type, const union codeview_type* type)
 {
     const union codeview_reftype* reftype = (const union codeview_reftype*)type;
-    int                 i, leaf_len, value;
+    int                 i, leaf_len, value = 0;
     unsigned int        j;
     const char*         str;
 
--- a/tools/winebuild/import.c
+++ b/tools/winebuild/import.c
@@ -182,8 +182,11 @@ static int func_cmp( const void *func1,
 {
     const ORDDEF *odp1 = *(const ORDDEF * const *)func1;
     const ORDDEF *odp2 = *(const ORDDEF * const *)func2;
-    return strcmp( odp1->name ? odp1->name : odp1->export_name,
-                   odp2->name ? odp2->name : odp2->export_name );
+    const char *name1 = odp1->name ? odp1->name : odp1->export_name;
+    const char *name2 = odp2->name ? odp2->name : odp2->export_name;
+    if ( !name1 || !name2 )
+      fatal_error("Unable to compare function names\n");
+    return strcmp( name1, name2 );
 }
 
 /* remove a name from a name table */
--- a/dlls/mciseq/mcimidi.c
+++ b/dlls/mciseq/mcimidi.c
@@ -246,9 +246,9 @@ static DWORD	MIDI_mciReadNextEvent(WINE_
 {
     BYTE	b1, b2 = 0, b3;
     WORD	hw = 0;
-    DWORD	evtPulse;
+    DWORD	evtPulse = 0;
     DWORD	evtLength;
-    DWORD	tmp;
+    DWORD	tmp = 0;
 
     if (mmioSeek(wmm->hFile, mmt->dwIndex, SEEK_SET) != mmt->dwIndex) {
 	WARN("Can't seek at %08X\n", mmt->dwIndex);
--- a/dlls/riched20/editor.c
+++ b/dlls/riched20/editor.c
@@ -1142,6 +1142,9 @@ static HRESULT insert_static_object(ME_T
       stgm.tymed = TYMED_GDI;
       stgm.u.hBitmap = hbmp;
       fm.cfFormat = CF_BITMAP;
+  } else {
+      stgm.tymed = TYMED_NULL;
+      ERR("using TYMED_NULL for fm\n");
   }
   stgm.pUnkForRelease = NULL;
 
--- a/dlls/sxs/name.c
+++ b/dlls/sxs/name.c
@@ -135,7 +135,6 @@ static HRESULT WINAPI name_GetDisplayNam
 {
     static const WCHAR fmtW[] = {',','%','s','=','\"','%','s','\"',0};
     struct name *name = impl_from_IAssemblyName( iface );
-    WCHAR version[30];
     unsigned int len;
 
     TRACE("%p, %p, %p, 0x%08x\n", iface, buffer, buflen, flags);
@@ -146,7 +145,7 @@ static HRESULT WINAPI name_GetDisplayNam
     if (name->arch)    len += lstrlenW( archW ) + lstrlenW( name->arch ) + 4;
     if (name->token)   len += lstrlenW( tokenW ) + lstrlenW( name->token ) + 4;
     if (name->type)    len += lstrlenW( typeW ) + lstrlenW( name->type ) + 4;
-    if (name->version) len += lstrlenW( versionW ) + lstrlenW( version ) + 4;
+    if (name->version) len += lstrlenW( versionW ) + lstrlenW( name->version ) + 4;
     if (len > *buflen)
     {
         *buflen = len;
--- a/dlls/msi/automation.c
+++ b/dlls/msi/automation.c
@@ -1837,7 +1837,7 @@ static HRESULT InstallerImpl_SummaryInfo
     UINT ret;
     HRESULT hr;
     MSIHANDLE hsuminfo;
-    IDispatch *dispatch;
+    IDispatch *dispatch = NULL;
     VARIANTARG varg0, varg1;
 
     if (!(wFlags & DISPATCH_PROPERTYGET))
--- a/dlls/msvcrt/time.c
+++ b/dlls/msvcrt/time.c
@@ -1998,6 +1998,8 @@ int CDECL _timespec32_get(struct _timesp
 {
     struct _timespec64 ts64;
 
+    if(base == 0)
+        return 0;
     if(_timespec64_get(&ts64, base) != base)
         return 0;
     if(ts64.tv_sec != (MSVCRT___time32_t)ts64.tv_sec)
