description: revert upstream msvcrt commit de5d147de8b3d9c2f65bcb6d5646320ef1d92e58

--- a/dlls/wininet/ftp.c
+++ b/dlls/wininet/ftp.c
@@ -3745,7 +3745,7 @@ static BOOL FTP_ParseNextFile(INT nSocke
 
             pszToken = strtok(NULL, szSpace);
             if(!pszToken) continue;
-            if(!stricmp(pszToken, "<DIR>")) {
+            if(!_strnicmp(pszToken, "<DIR>", -1)) {
                 lpfp->bIsDirectory = TRUE;
                 lpfp->nSize = 0;
                 TRACE("Is directory\n");
--- a/dlls/wininet/internet.c
+++ b/dlls/wininet/internet.c
@@ -2386,7 +2386,7 @@ static BOOL is_domain_suffix( const char
     int len_domain = strlen( domain ), len_suffix = strlen( suffix );
 
     if (len_suffix > len_domain) return FALSE;
-    if (!stricmp( domain + len_domain - len_suffix, suffix )) return TRUE;
+    if (!_strnicmp( domain + len_domain - len_suffix, suffix, -1 )) return TRUE;
     return FALSE;
 }
 
