description: revert upstream msvcrt commit 9fe900532eb310d114c04d15a8d0b986ec831552

--- a/dlls/wininet/internet.c
+++ b/dlls/wininet/internet.c
@@ -4263,10 +4263,9 @@ static BOOL calc_url_length(LPURL_COMPON
 
         if (!url_uses_default_port(nScheme, lpUrlComponents->nPort))
         {
-            WCHAR port[MAX_WORD_DIGITS + 1];
+            char szPort[MAX_WORD_DIGITS+1];
 
-            _ltow(lpUrlComponents->nPort, port, 10);
-            *lpdwUrlLength += lstrlenW(port);
+            *lpdwUrlLength += sprintf(szPort, "%d", lpUrlComponents->nPort);
             *lpdwUrlLength += strlen(":");
         }
 
@@ -4422,8 +4421,10 @@ BOOL WINAPI InternetCreateUrlW(LPURL_COM
 {
     DWORD dwLen;
     INTERNET_SCHEME nScheme;
+    WCHAR *start = lpszUrl;
 
     static const WCHAR slashSlashW[] = {'/','/'};
+    static const WCHAR fmtW[] = {'%','u',0};
 
     TRACE("(%p,%d,%p,%p)\n", lpUrlComponents, dwFlags, lpszUrl, lpdwUrlLength);
 
@@ -4508,9 +4509,9 @@ BOOL WINAPI InternetCreateUrlW(LPURL_COM
 
         if (!url_uses_default_port(nScheme, lpUrlComponents->nPort))
         {
-            *lpszUrl++ = ':';
-            _ltow(lpUrlComponents->nPort, lpszUrl, 10);
-            lpszUrl += lstrlenW(lpszUrl);
+            *lpszUrl = ':';
+            lpszUrl++;
+            lpszUrl += swprintf(lpszUrl, *lpdwUrlLength - (lpszUrl - start), fmtW, lpUrlComponents->nPort);
         }
 
         /* add slash between hostname and path if necessary */
