description: adjust search paths for addon installers
author: Michael Gilbert <mgilbert@debian.org>
author: Jens Reyer <jre.winesim@gmail.com>

--- a/dlls/appwiz.cpl/addons.c
+++ b/dlls/appwiz.cpl/addons.c
@@ -212,7 +212,8 @@ static enum install_res install_from_dos
 
     lstrcpyW( path, dir );
     if (!wcsncmp( path, ntprefixW, wcslen(ntprefixW) ))  path[1] = '\\';  /* change \??\ into \\?\ */
-    if (len && path[len-1] != '/' && path[len-1] != '\\') path[len++] = '\\';
+    /* don't append "\\" to "\\usr\\share\\wine-", we directly append the subdir_name for packaged installers*/
+    if (len && path[len-1] != '/' && path[len-1] != '\\' && path[len-1] != '-') path[len++] = '\\';
 
     if (*subdir)
     {
@@ -319,10 +320,14 @@ static enum install_res install_from_def
     }
 
     if (ret == INSTALL_NEXT)
-        ret = install_from_unix_file(INSTALL_DATADIR "/wine/", addon->subdir_name, addon->file_name);
+        /* our DATDIR "usr/share/wine$(VERSION)" already defines the wine subdir */
+        ret = install_from_unix_file(INSTALL_DATADIR "/", addon->subdir_name, addon->file_name);
     if (ret == INSTALL_NEXT && strcmp(INSTALL_DATADIR, "/usr/share"))
         ret = install_from_unix_file("/usr/share/wine/", addon->subdir_name, addon->file_name);
     if (ret == INSTALL_NEXT)
+        /* packaged addon installers */
+        ret = install_from_unix_file("/usr/share/wine-", addon->subdir_name, addon->file_name);
+    if (ret == INSTALL_NEXT)
         ret = install_from_unix_file("/opt/wine/", addon->subdir_name, addon->file_name);
     return ret;
 }
