description: do not generate duplicate winapploader tools
author: Michael Gilbert <mgilbert@debian.org>
forwarded: not-needed

--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -3798,11 +3798,6 @@ static void output_sources( struct makef
         for (arch = 0; arch < archs.count; arch++) if (is_multiarch( arch )) output_module( make, arch );
         if (make->unixlib) output_unix_lib( make );
         if (make->importlib) for (arch = 0; arch < archs.count; arch++) output_import_lib( make, arch );
-        if (make->is_exe && !make->is_win16 && *dll_ext && strendswith( make->module, ".exe" ))
-        {
-            char *binary = replace_extension( make->module, ".exe", "" );
-            add_install_rule( make, binary, 0, "wineapploader", strmake( "t$(bindir)/%s", binary ));
-        }
     }
     else if (make->testdll)
     {
