description: don't append wine to the default path settings
author: Michael Gilbert <mgilbert@debian.org>
forwarded: not-needed

--- a/configure.ac
+++ b/configure.ac
@@ -94,9 +94,9 @@ AC_ARG_WITH(wine64,    AS_HELP_STRING([-
 
 AC_CANONICAL_HOST
 
-AC_SUBST(dlldir,[\${libdir}/wine])
-AC_SUBST(fontdir,[\${datadir}/wine/fonts])
-AC_SUBST(nlsdir,[\${datadir}/wine/nls])
+AC_SUBST(dlldir,[\${libdir}])
+AC_SUBST(fontdir,[/usr/share/wine/fonts])
+AC_SUBST(nlsdir,[\${datadir}/nls])
 AC_SUBST(srcdir)
 
 dnl **** Check for some programs ****
--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -2918,7 +2918,7 @@ static void output_source_in( struct mak
     output( "%s:", obj_dir_path( make, obj ));
     output_filenames( source->dependencies );
     output( "\n" );
-    add_install_rule( make, obj, xstrdup( obj ), strmake( "d$(datadir)/wine/%s", obj ));
+    add_install_rule( make, obj, xstrdup( obj ), strmake( "d$(datadir)/%s", obj ));
 }
 
 
--- a/dlls/ntdll/Makefile.in
+++ b/dlls/ntdll/Makefile.in
@@ -75,4 +75,4 @@ unix_loader_EXTRADEFS = \
 	-DBINDIR=\"${bindir}\" \
 	-DSYSTEMDLLPATH=\"${system_dllpath}\" \
 	-DDLL_TO_BINDIR=\"`${MAKEDEP} -R ${dlldir} ${bindir}`\" \
-	-DBIN_TO_DATADIR=\"`${MAKEDEP} -R ${bindir} ${datadir}/wine`\"
+	-DBIN_TO_DATADIR=\"`${MAKEDEP} -R ${bindir} ${datadir}`\"
--- a/libs/wine/Makefile.in
+++ b/libs/wine/Makefile.in
@@ -93,7 +93,7 @@ config_EXTRADEFS = \
 	-DLIB_TO_BINDIR=\"`${MAKEDEP} -R ${libdir} ${bindir}`\" \
 	-DLIB_TO_DLLDIR=\"`${MAKEDEP} -R ${libdir} ${dlldir}`\" \
 	-DBIN_TO_DLLDIR=\"`${MAKEDEP} -R ${bindir} ${dlldir}`\" \
-	-DBIN_TO_DATADIR=\"`${MAKEDEP} -R ${bindir} ${datadir}/wine`\"
+	-DBIN_TO_DATADIR=\"`${MAKEDEP} -R ${bindir} ${datadir}`\"
 
 SHAREDLIB = $(LIBWINE_SHAREDLIB)
 
