description: use host CFLAGS as the CFLAGS for mingw
author: Michael Gilbert <mgilbert@debian.org>
forwarded: not-needed

--- a/configure.ac
+++ b/configure.ac
@@ -878,12 +878,11 @@ then
         ac_save_CC="$CC"
         saved_CFLAGS=$CFLAGS
         saved_LDFLAGS=$LDFLAGS
-        AC_SUBST(CROSSCFLAGS,${CROSSCFLAGS:-"-g -O2"})
+        AC_SUBST(CROSSCFLAGS,${CFLAGS})
         AC_SUBST(EXTRACROSSCFLAGS,"-D__WINE_PE_BUILD -Wall")
         AC_SUBST(CROSSLDFLAGS)
 
         CC="$CROSSCC"
-        CFLAGS=$CROSSCFLAGS
         LDFLAGS=$CROSSLDFLAGS
         AC_CACHE_CHECK([whether $CROSSCC works], wine_cv_crosscc,
             [AC_COMPILE_IFELSE([AC_LANG_PROGRAM([])],[wine_cv_crosscc=yes],[wine_cv_crosscc=no])])
