description: disable obsolete wine_fold_string api
author: Michael Gilbert <mgilbert@debian.org>

--- a/libs/wine/Makefile.in
+++ b/libs/wine/Makefile.in
@@ -71,7 +71,6 @@ C_SRCS = \
 	cpsymbol.c \
 	cptable.c \
 	debug.c \
-	digitmap.c \
 	fold.c \
 	ldt.c \
 	loader.c \
--- a/libs/wine/fold.c
+++ b/libs/wine/fold.c
@@ -24,6 +24,7 @@
 
 #include "wine/unicode.h"
 
+#if 0
 static inline WCHAR to_unicode_digit( WCHAR ch )
 {
     extern const WCHAR wine_digitmap[] DECLSPEC_HIDDEN;
@@ -36,6 +37,7 @@ static inline WCHAR to_unicode_native( W
     extern const WCHAR wine_compatmap[] DECLSPEC_HIDDEN;
     return ch + wine_compatmap[wine_compatmap[ch >> 8] + (ch & 0xff)];
 }
+#endif
 
 static const WCHAR wine_ligatures[] =
 {
@@ -119,6 +121,7 @@ static inline const WCHAR* get_ligature(
     return empty_ligature;
 }
 
+#if 0
 /* fold a unicode string */
 int wine_fold_string_obsolete( int flags, const WCHAR *src, int srclen, WCHAR *dst, int dstlen )
 {
@@ -203,5 +206,6 @@ int wine_fold_string_obsolete( int flags
 }
 
 __ASM_OBSOLETE(wine_fold_string);
+#endif
 
 #endif  /* __ASM_OBSOLETE */
