description: disable upstream's font installation rules
author: Michael Gilbert <mgilbert@debian.org>

--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -2792,7 +2792,6 @@ static void output_source_sfd( struct ma
     }
     if (source->file->flags & FLAG_INSTALL)
     {
-        add_install_rule( make, source->name, ttf_obj, strmake( "D$(fontdir)/%s", ttf_obj ));
         output_srcdir_symlink( make, ttf_obj );
     }
 
