description: disable upstream's font installation rules
author: Michael Gilbert <mgilbert@debian.org>

--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -2811,13 +2811,13 @@ static void output_source_sfd( struct ma
                 fontforge, root_src_dir_path( "fonts/genttf.ff" ), source->filename );
         if (!(source->file->flags & FLAG_SFD_FONTS)) strarray_add( &make->font_files, ttf_obj );
     }
-    if (source->file->flags & FLAG_INSTALL)
+    if (0)
     {
         add_install_rule( make, source->name, ttf_obj, strmake( "D$(fontdir)/%s", ttf_obj ));
         output_srcdir_symlink( make, ttf_obj );
     }
 
-    if (source->file->flags & FLAG_SFD_FONTS)
+    if (0)
     {
         struct strarray *array = source->file->args;
 
