description: remove dependencies on prebuilt ttf font files
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/dwrite/tests/Makefile.in
+++ b/dlls/dwrite/tests/Makefile.in
@@ -8,5 +8,3 @@ C_SRCS = \
 
 FONT_SRCS = \
 	wine_test.sfd
-
-RC_SRCS = resource.rc
--- a/dlls/gdi32/tests/Makefile.in
+++ b/dlls/gdi32/tests/Makefile.in
@@ -21,5 +21,3 @@ FONT_SRCS = \
 	wine_longname.sfd \
 	wine_test.sfd \
 	wine_vdmx.sfd
-
-RC_SRCS = resource.rc
--- a/dlls/gdiplus/tests/Makefile.in
+++ b/dlls/gdiplus/tests/Makefile.in
@@ -15,7 +15,5 @@ C_SRCS = \
 	region.c \
 	stringformat.c
 
-RC_SRCS = resource.rc
-
 FONT_SRCS = \
 	wine_longname.sfd
