description: ignore vulkansc api
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/winevulkan/make_vulkan
+++ b/dlls/winevulkan/make_vulkan
@@ -607,6 +607,8 @@ class VkFunction(object):
 
         params = []
         for param in command.findall("param"):
+            if param.get("api") == "vulkansc":
+                continue
             vk_param = VkParam.from_xml(param, types, params)
             params.append(vk_param)
 
@@ -1947,6 +1949,8 @@ class VkStruct(Sequence):
 
         s = VkStruct(name, [], returnedonly, structextends, union=union)
         for member in struct.findall("member"):
+            if member.get("api") == "vulkansc":
+                continue
             vk_member = VkMember.from_xml(member, returnedonly, s)
             s.members.append(vk_member)
 
@@ -3555,6 +3559,8 @@ class VkRegistry(object):
         """ Parse the feature section, which describes Core commands and types needed. """
 
         for feature in root.findall("./feature"):
+            if feature.attrib["api"] == "vulkansc":
+                continue
             feature_name = feature.attrib["name"]
             for require in feature.findall("require"):
                 LOGGER.info("Including features for {0}".format(require.attrib.get("comment")))
@@ -3591,6 +3597,8 @@ class VkRegistry(object):
 
         alias_types = []
         for t in types:
+            if t.attrib.get("api") == "vulkansc":
+                continue
             type_info = {}
             type_info["category"] = t.attrib.get("category", None)
             type_info["requires"] = t.attrib.get("requires", None)
