description: fix mismatched types (-Wincompatible-pointer-types)
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/winegstreamer/media_source.c
+++ b/dlls/winegstreamer/media_source.c
@@ -598,7 +598,7 @@ static DWORD CALLBACK read_thread(void *
     struct media_source *source = arg;
     IMFByteStream *byte_stream = source->byte_stream;
     size_t buffer_size = 4096;
-    uint64_t file_size;
+    QWORD file_size;
     void *data;
 
     if (!(data = malloc(buffer_size)))
@@ -1393,7 +1393,7 @@ static HRESULT media_source_constructor(
     UINT64 total_pres_time = 0;
     struct wg_parser *parser;
     DWORD bytestream_caps;
-    uint64_t file_size;
+    QWORD file_size;
     unsigned int i;
     HRESULT hr;
 
--- a/programs/winedbg/be_arm.c
+++ b/programs/winedbg/be_arm.c
@@ -1555,12 +1555,12 @@ void be_arm_disasm_one_insn(ADDRESS64 *a
     int matched = 0;
 
     char tmp[64];
-    DWORD_PTR* pval;
+    union { DWORD dvalue; struct dbg_lvalue lvalue; } values;
 
-    if (!memory_get_register(CV_ARM_CPSR, &pval, tmp, sizeof(tmp)))
+    if (!memory_get_register(CV_ARM_CPSR, &values.lvalue, tmp, sizeof(tmp)))
         dbg_printf("\n\tmemory_get_register failed: %s", tmp);
     else
-        db_disasm_thumb = (*pval & 0x20) != 0;
+        db_disasm_thumb = (values.dvalue & 0x20) != 0;
 
     db_display = display;
 
