description: build with gcc instead of clang on arm64
author: Michael Gilbert <mgilbert@debian.org>

--- a/configure.ac
+++ b/configure.ac
@@ -997,11 +997,6 @@ do
     CFLAGS=$saved_CFLAGS
 done
 
-if test $HOST_ARCH = aarch64 -a "x$PE_ARCHS" = x
-then
-    AC_MSG_ERROR([PE cross-compilation is required for ARM64, please install clang/llvm-dlltool/lld, or llvm-mingw.])
-fi
-
 WINE_NOTICE_WITH(mingw,[test "x$PE_ARCHS" = "x"],
                  [MinGW compiler not found, cross-compiling PE files won't be supported.])
 
