description: pcap.h defines PCAP_OPENFLAG_PROMISCUOUS starting with 1.9.0
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/wpcap/wpcap.c
+++ b/dlls/wpcap/wpcap.c
@@ -254,7 +254,9 @@ int CDECL wine_pcap_next_ex(pcap_t *p, s
     return pcap_next_ex(p, pkt_header, pkt_data);
 }
 
+#ifndef PCAP_OPENFLAG_PROMISCUOUS
 #define PCAP_OPENFLAG_PROMISCUOUS 1
+#endif
 
 pcap_t* CDECL wine_pcap_open(const char *source, int snaplen, int flags, int read_timeout,
                              void *auth, char *errbuf)
