description: add Makefile rules for building just the manpages
author: Michael Gilbert <mgilbert@debian.org>

--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -3657,6 +3657,7 @@ static void output_sources( struct makef
             output_filenames_obj_dir( make, make->all_targets );
             output( "\n" );
         }
+        output( "install-man:: $(shell echo ${MANPAGES} | sed s/\\\\.in//g)\n" );
         output_install_rules( make, INSTALL_LIB, "install-lib" );
         output_install_rules( make, INSTALL_DEV, "install-dev" );
         output_uninstall_rules( make );
