description: build using system zlib
author: Michael Gilbert <mgilbert@debian.org>
forwarded: not-needed

--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -1454,6 +1454,7 @@ static struct file *open_include_file( c
     if (pFile->type == INCL_SYSTEM && pFile->use_msvcrt &&
         !make->extlib && !pFile->included_by->is_external)
     {
+        if (!strcmp( pFile->name, "zlib.h" )) return NULL;
         if (!strcmp( pFile->name, "stdarg.h" )) return NULL;
         if (!strcmp( pFile->name, "x86intrin.h" )) return NULL;
         if (has_external_import( make )) return NULL;
--- a/configure.ac
+++ b/configure.ac
@@ -3525,7 +3525,6 @@ WINE_CONFIG_MAKEFILE(libs/wine)
 WINE_CONFIG_MAKEFILE(libs/wmcodecdspuuid)
 WINE_CONFIG_MAKEFILE(libs/xml2)
 WINE_CONFIG_MAKEFILE(libs/xslt)
-WINE_CONFIG_MAKEFILE(libs/zlib)
 WINE_CONFIG_MAKEFILE(loader)
 WINE_CONFIG_MAKEFILE(nls)
 WINE_CONFIG_MAKEFILE(po)
